packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -192,12 +192,15 @@ class ArtifactStore {
     }
   }
 
-  static void ensureHasSkyEnginePackage() {
-    Directory skyEnginePackage = new Directory(path.join(packageRoot, 'sky_engine'));
-    if (!skyEnginePackage.existsSync()) {
+  static void validateSkyEnginePackage() {
+    if (engineRevision == null) {
       printError("Cannot locate the sky_engine package; did you include 'flutter' in your pubspec.yaml file?");
       throw new ProcessExit(2);
     }
+    if (engineRevision != expectedEngineRevision) {
+      printError("Error: incompatible sky_engine package; please run 'pub get flutter' to get the correct one.\n");
```

Comment:
I don't think we want to require devs to upgrade their flutter (I expect that intentional flutter downgrades will be a common scenario that causes this error to be displayed). However the "flutter" part of "pub get flutter" seems to be ignored so I'll omit that.